react-workshop/src/components/news.js

```diff
@@ -0,0 +1,124 @@
+import React, { Component } from "react";
+import NewsItem from "./news-item";
+import SearchBar from "./search-bar";
+import Button from "./button";
+
+class NewsList extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      search: "",
+      list: props.list
+    };
+  }
+
+  showAlert() {
+    alert("Moggu homo");
+  }
+
+  onChangeSearch = search => this.setState({ search });
+
+  addNews(event) {
+    event.preventDefault();
+    let title = this.refs.title.value;
+    let author = this.refs.author.value;
+
+    let objectID = Math.floor(Math.random() * 100 + 1);
+
+    this.setState({
+      list: this.state.list.concat({
+        objectID,
+        title,
+        author
+      })
+    });
+    this.refs.title.value = "";
+    this.refs.author.value = "";
+  }
+
+  onDelete(objectID) {
+    this.setState({
+      list: this.state.list.filter(listItem => listItem.objectID !== objectID)
+    });
+  }
+
+  toggleDiv = () => {
+    const { show } = this.state;
+    this.setState({ show: !show });
```

Comment:
voor togglen zou je https://reactjs.org/docs/state-and-lifecycle.html -> `State Updates May Be Asynchronous` door kunnen lezen. dan zie je hoe je regel 47 en 48 samen kan voegen en waarom dat beter is